### PR TITLE
Working towards an automatic Fore/Aft/Mid L1 processing chain

### PR DESCRIPTION
**Pull request by DavidMcCann-NOC**
- Updates to how we handle the antenna azimuth to include the Mid antenna and to be more robust
- Functions to assist with loading in folders of .nc files and identify antenna triplets (fore/aft/mid files) for automated L1 processing

### CODE REVIEW

File: seastar/oscar/level1.py [hunk 1]
```diff
@@ -103,12 +103,11 @@ def add_antenna_baseline(ds, baseline):
         Antenna baseline distance (m)
 
     """
-    if 'Baseline' not in ds.data_vars:
-        ds['Baseline'] = baseline
-        ds.Baseline.attrs['long_name'] =\
-            'Antenna ATI baseline'
-        ds.Baseline.attrs['units'] = '[m]'
-    return ds
+    Baseline = xr.DataArray(data=baseline)
```
Comment (ACHMartin) [inline seastar/oscar/level1.py:106]: I think the PEP8 required a variable NOT to start with a capsize letter

File: seastar/oscar/level1.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def check_antenna_polarization(ds):
     return ds
 
 
-def add_antenna_baseline(ds, baseline):
+def add_antenna_baseline(baseline):
```
Comment (ACHMartin) [inline seastar/oscar/level1.py:87]: should we change the name from add_antenna.. to compute_antenna.. as nothing is added anymore

File: seastar/oscar/level1.py [hunk 3]
```diff
@@ -476,55 +479,26 @@ def compute_antenna_azimuth_direction(ds, antenna):
...
-                                                   + 135,
-                                                   360)
+    look_direc_angle = {'L': -90, 'R': 90}
+
+    if 'SquintImage' in ds:
+        ds['AntennaAzimuthImage'] = (np.mod(
+            ds.OrbitHeadingImage
+            + look_direc_angle[lookdirec],
+            360)
+            + ds.SquintImage)
```
Comment (ACHMartin) [inline seastar/oscar/level1.py:491]: very nice, much compacter and easier to read

File: seastar/utils/tools.py [hunk 4]
```diff
@@ -322,3 +323,62 @@ def  wgs2utm_v3(lat, lon, utmzone, utmhemi):
                                               (61 - 58 * T + T ** 2 + 600 * C -
                                                330 * eps) * A ** 6/720)
     return x, y
+
+
+def indentify_antenna_location(ds):
```
Comment (ACHMartin) [inline seastar/utils/tools.py:328]: typo on the name of the function

File: seastar/utils/tools.py [hunk 5]
```diff
@@ -322,3 +323,62 @@ def  wgs2utm_v3(lat, lon, utmzone, utmhemi):
...
+        antenna_location = 'Mid'
+    elif doppler_mean < 0:
+        antenna_location = 'Aft'
+    elif doppler_mean > 0:
+        antenna_location = 'Fore'
+
+    return antenna_location
+
+
+def list_duplicates(seq):
```
Comment (ACHMartin) [inline seastar/utils/tools.py:358]: when do we need to use this?

**Comment by ACHMartin** [comment]
good to go for me after the minor changes

File: seastar/utils/tools.py (hunk 5 above)
Comment (DavidMcCann-NOC) [reply seastar/utils/tools.py:358]: Its coming in the next commits - its part of a framework i've built to address #167, which will standardise loading in the .nc files as Fore/Aft/Mid triplets - this little function creates a handy generator that can be used to very efficiently hunt for duplicates in a sequence of data. In our case the acquisition times in a (potentially long) file list of OSCAR data

File: seastar/utils/tools.py (hunk 4 above)
Comment (DavidMcCann-NOC) [reply seastar/utils/tools.py:328]: Well caught!

File: seastar/oscar/level1.py (hunk 3 above)
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:491]: Thanks! I think so too. Its been made much easier with the new way I'm loading in data (next commits), sort of all fell in to place. No need to specify antenna direction manually anymore

File: seastar/oscar/level1.py (hunk 2 above)
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:87]: I agree

File: seastar/oscar/level1.py (hunk 1 above)
Comment (DavidMcCann-NOC) [reply seastar/oscar/level1.py:106]: Very true - strangely my PEP8 linting didn't catch that. I must have been messing around in Matlab too much recently...

**Pull request merged**